File: sources/com/s8/fwks/palm/structs/columns/Column.java
package com.s8.fwks.palm.structs.columns;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;
import com.s8.fwks.palm.PalmWebSources;


/**
 * 
 * @author pierreconvert
 *
 */
public class Column extends S8WebObject {

	
	/**  */
	public final static String WEBPATH = PalmWebSources.ROOT_WEBPATH + "/structs/columns/Column";
	
	
	/**
	 * 
	 * @param branch
	 */
	public Column(S8WebFront branch) {
		super(branch, WEBPATH);
	}
	
	
	
	/**
	 * 
	 * @param name
	 */
	public void setElements(S8WebObject... elements) {
		vertex.outbound().setObjectListField("elements", elements);
	}
	
	
	public void setWidth(double width) {
		vertex.outbound().setFloat32Field("width", (float) width);
	}
	
}

File: web-sources/S8-fwks-palm/structs/columns/Column.js

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";
import { S8Object } from "/S8-api/S8Object.js";

S8WebFront.CSS_import('/S8-fwks-palm/structs/columns/Columns.css');


/**
 * 
 */
export class Column extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;
    
    /**
     * 
     */
    constructor(){
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("palm-struct-column");

        this.contentNode = document.createElement("div");
        this.wrapperNode.appendChild(this.contentNode);

    }


    /**
     * 
     * @returns 
     */
    getEnvelope(){
        return this.wrapperNode;
    }

    /** @param{S8Object[]} buffer */
    getComponents(buffer) {
        if(this.elements) { buffer.push(this.elements); }
    }

    /**
     * @param{string} eventType
     */
    onDownwardEvent(eventType){
       
    }


    /** @param {S8Object[]} elements */
    S8_set_elements(elements){
       S8WebFront.setS8Elements(this.contentNode, elements);
    }


    /**
     * 
     * @param {number} width 
     */
    S8_set_width(width){
        if(width < 0){ /* clear width */
            this.wrapperNode.style.removeProperty("width");
        }
        else if(width > 0){
            this.wrapperNode.style.width = `${width * 100}vw`;
        }
    }


    /** render */
    S8_render(){}


    /** dispose */
    S8_dispose(){}

}